File: fix_jsx.py
import os

file_path = r'c:\Users\Dell\Desktop\Captina_app_new\src\app\booking\page.js'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Fix the map closure and surrounding divs
# The broken part looks like:
#                                                 </div>
#                                             </div>
#                                         </div>
#                                     </div>
# Looking for the pattern of the map start and then the broken end.
# We want to replace the sequence of 4 divs with the correct closure.

old_block = """                                                </div>
                                            </div>
                                        </div>
                                    </div>"""

new_block = """                                                </div>
                                            ))}
                                        </div>
                                    </div>"""

if old_block in content:
    content = content.replace(old_block, new_block, 1)
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    print("Successfully fixed the map closure.")
else:
    print("Could not find the target block. Checking for slightly different spacing...")
    # Try a more flexible match if necessary, but the provided block should match based on view_file.

File: fix_jsx.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('c:', 'Users', 'Dell', 'Desktop', 'Captina_app_new', 'src', 'app', 'booking', 'page.js');
let content = fs.readFileSync(filePath, 'utf8');

// The problematic map closure
const oldPart = /\{typeof item\.value === 'object' \? getText\(item\.value\) : item\.value\}\s+<\/p>\s+<\/div>\s+<\/div>\s+<\/div>\s+<\/div>\s+<\/div>/;

if (oldPart.test(content)) {
    console.log("Found broken map closure. Fixing...");
    
    const replacement = `{typeof item.value === 'object' ? getText(item.value) : item.value}
                                                        </p>
                                                    </div>
                                                </div>
                                            ))}
                                        </div>
                                    </div>`;
                                    
    content = content.replace(oldPart, replacement);
    fs.writeFileSync(filePath, content, 'utf8');
    console.log("Fixed successfully.");
} else {
    console.log("Could not find the target section using regex.");
    // Try a more flexible search
    console.log("Trying to find 'item.value' and its surrounding divs...");
    const flexibleMatch = content.indexOf("{typeof item.value === 'object' ? getText(item.value) : item.value}");
    if (flexibleMatch !== -1) {
        console.log("Found 'item.value' at index", flexibleMatch);
        // Let's just find the next 5 </div> tags and replace that section.
    }
}
